src/main/java/com/johan/blignaut/entelect/vitalityincubator/helpers/GeoDBMappingHelper.java

```diff
@@ -15,40 +15,54 @@
 import retrofit2.Response;
 
 import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
 
 @Component
 public class GeoDBMappingHelper {
 
-    private final GeodbApi geodbApi;
-    private final GeoDBMappingService geoDBMappingService;
+    private final ThreadPoolExecutor executor;
+    private GeodbApi geodbApi;
+    private GeoDBMappingService geoDBMappingService;
 
     @Value("${itemsperpage}")
     private int itemsPerPage;
 
     private static final String PLACES = "places";
 
-    public GeoDBMappingHelper(GeodbApi geodbApi, GeoDBMappingService geoDBMappingService) {
+    public GeoDBMappingHelper(GeodbApi geodbApi, GeoDBMappingService geoDBMappingService, ThreadPoolExecutor executor) {
         this.geodbApi = geodbApi;
         this.geoDBMappingService = geoDBMappingService;
+        this.executor = executor;
     }
 
     @Cacheable(PLACES)
     public GeoDBServer mapClientToServerGeoDB(String filterByName) {
         try {
-            Call<GeoDBClient> call = geodbApi.getPlaces(itemsPerPage);
-            Response<GeoDBClient> response = call.execute();
-            if (response.isSuccessful()) {
-                GeoDBClient geodbClient = response.body();
-                return geoDBMappingService.mapClientToServerGeoDB(geodbClient, filterByName);
+            Future<GeoDBServer> geoDBServerFuture = executor.submit(() -> {
+                try {
```

Comment:
Please set the `RequestAttributes` on the new thread created by the Future.
The RequestAttributes have important info about the incoming request and, while probably not very relevant in your case, not having the correct `RequestAttributes` available has been known to cause bugs in our microservices